src/main/java/org/traccar/protocol/HuabaoProtocolDecoder.java

```diff
@@ -699,6 +699,16 @@ private Position decodeLocation(DeviceSession deviceSession, ByteBuf buf) {
                         }
                     }
                     break;
+                case 0xF6:
+                    buf.readUnsignedByte(); // data type
+                    int fieldMask = buf.readUnsignedByte();
+                    if (BitUtil.check(fieldMask, 0)) {
+                        buf.readUnsignedShort(); // light
+                    }
+                    if (BitUtil.check(fieldMask, 1)) {
+                        position.set(Position.PREFIX_TEMP + 1, buf.readUnsignedShort() * 0.1);
```

Comment:
Can this temperature be negative? Can we add a test case?

Reply:
Probably yes, you mean a test case with negative temperature?

Reply:
Yes.

Reply:
I'll put the device in the freezer to test.

Reply:
it works:

![Captura de ecrã 2024-08-25 à(s) 00 56 38](https://github.com/user-attachments/assets/0a23596f-19c1-45c7-9e96-3a82a76fc473)